### PR TITLE
Cleanup schema

### LINKED ISSUE #2326: [DCR] [Adaptive] Re-evaluate all events exposed via Adaptive dialog including protocol, dialog and adaptive events.
Filing this ticket so we review and distill the set of events exposed via adaptive. The current set is a union of protocol level events, dialog events as well as adaptive events. we should pare back the list to only those that are backed by actual user scenarios as well as consider adding new events that help fulfill core unmet user needs.

**Adaptive events** 

| Event name         | Description/ purpose/ use case                                                      | Payload         |
|--------------------|-------------------------------------------------------------------------------------|-----------------|
| [ ] BeginDialog        | The active dialog has started.                                                      | N/A             |
| [ ] ActivityRecieved   | BF activity recieved. Use this for fine grained control of specific BF activity type | turn.activity   |
| [ ] RecognizedIntent   | The active dialog's recognizer has new information.                                 | turn.recognized |
| [ ] UnknownIntent      | The active dialog's recognizer did not detect an intent.                            | N/A             |
| [ ] CancelDialog       | The active dialog has been cancelled.                                               | N/A             |
| [ ] [REMOVE] EndDialog   | The active dialog has ended.                                                        | N/A             |
| [ ] RePromptDialog     |  Emitted when `DialogContext.repromptDialog()` is called. |  N/A               |
| [ ] [REMOVE] ResumeDialog       | ?                                                                                   | ?               |
| [ ] [REMOVE] SequenceEnded      | ?                                                                                   | ?               |
| [ ] [REMOVE] SequenceStarted    | ?                                                                                   | ?               |
| [ ] Error  | When an exception occurs it results in a call to `cancelAllDialogs()` with an "error" event   |  Exception raised| 

**Event handlers** 

| Name                                | Description                          | Base event       | Constraint                             | Use case | Keep ? |
|-------------------------------------|--------------------------------------|------------------|----------------------------------------|----------|--------|
| [ ] OnEvent                             | Generic event handler                |                  |                                        |          |        |
| [ ] OnIntent                            | Intent event handler                 | RecognizedIntent | turn.recognized.intent == 'IntentName' |          |        |
| [ ] OnUnknownIntent                     | Unknown intent event handler         | UnknownIntent    |                                        |          |        |
| [ ] OnBeginDialog                       | Begin dialog handler                 | BeginDialog      | N/A                                    |          |        |
| [ ] OnActivity                       \/ | Generic activity handler             | ActivityRecieved | N/A                                    |          |        |
| [ ] OnConversationUpdateActivity        | Conversation update activity handler | ActivityRecieved | ActivityTypes.ConversationUpdate       |          |        |
| [ ] OnEndOfConversationActivity         | End of conversation activity handler | ActivityRecieved | ActivityTypes.EndOfConversation        |          |        |
| [ ] OnEventActivity                     | Event activity handler               | ActivityRecieved | ActivityTypes.Event                    |          |        |
| [ ] OnHandoffActivity                   | Hand off activity handler            | ActivityRecieved | ActivityTypes.Handoff                  |          |        |
| [ ] OnInvokeActivity                    | Invoke activity handler              | ActivityRecieved | ActivityTypes.Invoke                   |          |        |
| [ ] OnMessageActivity                   | Message activity handler             | ActivityRecieved | ActivityTypes.Message                  |          |        |
| [ ] OnMessageDeletionActivity           | Message deletion activity handler    | ActivityRecieved | ActivityTypes.MessageDeletion          |          |        |
| [ ] OnMessageReactionActivity           | Message reaction activity handler    | ActivityRecieved | ActivityTypes.MessageReaction          |          |        |
| [ ] OnMessageUpdateActivity             | Message update activity handler      | ActivityRecieved | ActivityTypes.MessageUpdate            |          |        |
| [ ] OnTypingActivity                    | Typing activity handler              | ActivityRecieved | ActivityTypes.Typing                   |          |        |

If ActivityRecieved is added, nothing else should fire since the event loop should stop for that turn. 
CancelDialog or EndDialog cannot include actions that prolong beyond the current turn (e.g. cannot hold an input action)
If multiple events are eligible to fire, the first one will win.

### PR DESCRIPTION
**Pull request by tomlm**
cleanup Events->Trigger in schema files
* another pass at normalization of names 
* Add OnCondition as root trigger condition instead of TriggerHandler
* Add OnCustomEvent per Vishwac
* Made DialogEvent.Type instead of .Types 
* refresh all schema files

Fixes #2326

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnCustomEvent.schema [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                "event": {
+                    "type": "string",
+                    "title": "Custom Event Name",
+                    "description": "Custom event name to match to trigger these actions"
+                }
+            }
+        }
+    ],
+    "required": [
+        "actions",
```
Comment (vishwacsena) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnCustomEvent.schema]: Where is this 'action' property defined? Should it not be listed under 'properties'?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnDialogEvent.schema [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                "event": {
+                    "type": "string",
+                    "title": "Dialog Event Name",
+                    "description": "Dialog event name to match to trigger these actions"
+                }
+            }
+        }
+    ],
+    "required": [
+        "actions",
```
Comment (vishwacsena) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnDialogEvent.schema]: Where is this 'action' property defined? Should it not be listed under 'properties'?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnDialogEvent.schema [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "description": "Defines a actions which are executed when a dialog event matching event is is true",
+    "type": "object",
+    "allOf": [
+        {
+            "$ref": "./Microsoft.OnCondition.schema#"
+        },
+        {
+            "properties": {
+                "event": {
+                    "type": "string",
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnDialogEvent.schema]: Should onDialogEvent's event is a enum and onCustomizedEvent's event is a free string. 

I thought onCustomEvent is for listening for customized event
and onDialogEvent is for listening for system event.

Comment (vishwacsena) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/TriggerConditions/Microsoft.OnDialogEvent.schema]: @tomlm was expecting to see something like this in the OnDialogEvent schema --
```json
"events": {
                    "type": "array",
                    "description": "Events to trigger this rule for",
                    "items": {
                        "type": "string",
                        "enum": [
                            "beginDialog",
                            "consultDialog",
                            "cancelDialog",
                            "activityReceived",
                            "recognizedIntent",
                            "unknownIntent",
                            "actionsStarted",
                            "actionsSaved",
                            "actionsEnded",
                            "actionsResumed"
                        ]
                    }
                }
```

**Pull request merged**